openwhisk/README.md

```diff
@@ -0,0 +1,262 @@
+istio-egressgateway-6ddf4c8bd6-2dxhc       1/1       Running     1          4d21h
+istio-galley-7dd996474-pdd6h               1/1       Running     1          4d21h
+istio-ingressgateway-84b89d647f-cxrwx      1/1       Running     1          4d21h
+istio-pilot-86bb4fcbbd-5ns5q               2/2       Running     0          4d21h
+istio-pilot-86bb4fcbbd-vd2xr               2/2       Running     0          4d21h
+istio-pilot-86bb4fcbbd-zstrw               2/2       Running     0          4d21h
+istio-policy-5c4d9ff96b-559db              2/2       Running     1          4d21h
+istio-security-post-install-dv8s7          0/1       Completed   0          4d21h
+istio-sidecar-injector-6977b5cf5b-94hj5    1/1       Running     0          4d21h
+istio-telemetry-7676df547f-jp952           2/2       Running     1          4d21h
+zipkin-8594bbfc6b-z6vdg                    1/1       Running     1          4d21h  
+```
+</details>
+
+#### Verify your default namespace uses Istio for all services
+
+Check the `default` namespace has the label **istio-injection** and it is set to **enabled**:
+
+```bash
+$ kubectl get namespace default -o yaml
+```
+
+Example output:
+```
+apiVersion: v1
+kind: Namespace
+metadata:
+  creationTimestamp: "2019-06-13T16:52:46Z"
+  labels:
+    istio-injection: enabled
+  name: default
+  resourceVersion: "18019"
+  selfLink: /api/v1/namespaces/default
+  uid: ab987afe-8dfb-11e9-8092-025000000001
+spec:
+  finalizers:
+  - kubernetes
+status:
+  phase: Active
+```
+
+**Note**: If you do not see the **istio-injection** label, verify you issued the 'kubectl' command to set this label to the default namespace. See [Troubleshooting](#troubleshooting) section for more information.
+
+# Building and Serving OpenWhisk Runtime Tekton Task 
+
+All OpenWhisk Runtime Tekton Task require a valid Kubernetes **Service Account** with access to a Kubernetes **Secret** that contains your username and password to image resgistry.  This credential will be used as part of the Tekton build process to "push" your Knative application image containing your OpenWhisk Action to image resgistry of your choice.
+
+## Clone this repository
+
+```bash
+$ git clone https://github.com/tektoncd/catalog.git
+$ cd openwhisk
+```
+
+## Register Secrets for Docker Hub
+
+Export your Docker Hub **username** and **password** required to register a new secret in Kubernetes.
+
+```
+$ export DOCKER_USERNAME=<your docker hub username>
+$ export DOCKER_PASSWORD=<your docker hub password>
+```
+
+Use the following `sed` command which will generate a `docker-secret.yaml` file from the file [docker-secret.yaml.tmpl](docker-secret.yaml.tmpl) and replace the **username** and **password** values with your Docker Hub username and password from the environment variables you exported above:
+
+```
+sed -e 's/${DOCKER_USERNAME}/'"$DOCKER_USERNAME"'/' -e 's/${DOCKER_PASSWORD}/'"$DOCKER_PASSWORD"'/' docker-secret.yaml.tmpl > docker-secret.yaml
```

Comment:
Nit: You could possibly do this with `kubectl create secret docker-registry` and `kubectl annotate secret` and remove the `.tmpl` file from the repo completely.